### PR TITLE
Health Metric using Chart.js

### LINKED ISSUE #4753: Display App Metrics
**What type(s) of user does this feature affect?**
Any Visitor

**Description**
Use [chart.js](https://www.chartjs.org/docs/latest/getting-started/installation.html) to create a [bubble chart](https://www.chartjs.org/docs/latest/charts/bubble.html) of when case case contacts  are created. 

Display the metrics at the `/health` page

1. Add chart.js to application.js
2. For the data points get a list of the `created_at` times for case contacts created in the last week. The data points are truncated by day for the y axis and truncated by hour for the x axis. If 2 data points overlap they are combined and the radius of the circle is increased. Time is fetched from the backend as unix time and converted to local time on the frontend. Do not make any other details of the case contacts accessible on the page.  
3. Create a chart like the one below. The y axis of the graph is the day of the week and the x axis is the hour of the day. 
image

See [this guide](https://github.com/rubyforgood/casa/wiki/Working-with-Frontend-Assets) for working with javascript in the CASA project.

_Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  
[Link to new theme site](https://casa-new-theme.herokuapp.com/languages/new)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin
- all_casa_admin1@example.com view site as an all casa admin
  - go to `/all_casa_admins/sign_in`  

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in slack https://rubyforgood.herokuapp.com/ #casa channel to ask questions quickly and hear about office hours (currently Tuesday 5-7pm Pacific), stakeholder news, and upcoming new issues.

### PR DESCRIPTION
**Pull request by aboongm**
### What github issue is this PR for, if any?
Resolves #4753

### What changed, and why?
- package.json
   - add chart
   - add luxon
   - chartjs-adapter-luxon
    
- case_contacts_controller.rb
   - add case_contacts_creation_times_in_last_week method
   
- routes.rb
   - provides route for case_contacts_creation_times_in_last_week

- application.js
    - require display_app_metric
    
- display_app_metric.js
    - This javascript file will use chart.js and create the bubble chart

- index.html.erb (health)
    - This will render the chart on browser

### How will this affect user permissions?
- NA

### How is this tested? (please write tests!) 💖💪
- health_spec.rb (requests test for metric method)

### Screenshots please :)

127 0 0 1_3000_health

### Feelings gif (optional)
What gif best describes your feeling working on this issue? https://giphy.com/
How to embed:
`alt text`

### Feedback please? (optional)
We are very interested in your feedback! Please give us some :) https://forms.gle/1D5ACNgTs2u9gSdh9

### CODE REVIEW

**Comment by compwron** [comment]
Please add screenshots

File: app/javascript/application.js [hunk 1]
```diff
@@ -4,6 +4,8 @@ import './jQueryGlobalizer.js'
 import 'bootstrap'
 import 'bootstrap-select'
 import './sweet-alert-confirm.js'
+// import { Chart, registerables } from 'chart.js'
```
Comment (compwron) [inline app/javascript/application.js:7]: remove commented out code

File: app/controllers/case_contacts_controller.rb [hunk 2]
```diff
@@ -134,6 +136,17 @@ def restore
     redirect_to request.referer
   end
 
+  def case_contacts_creation_times_in_last_week
+    # Get the case contacts created in the last week
+    case_contacts = CaseContact.where("created_at >= ?", 10.week.ago)
```
Comment (compwron) [inline app/controllers/case_contacts_controller.rb:141]: metrics are much more valuable per org- are we planning to split these out per org in a later PR?

File: app/controllers/case_contacts_controller.rb [hunk 3]
```diff
@@ -5,6 +5,8 @@ class CaseContactsController < ApplicationController
   before_action :set_contact_types, only: %i[new edit update create]
   before_action :require_organization!
   after_action :verify_authorized
+  skip_before_action :authenticate_user!, only: :case_contacts_creation_times_in_last_week
```
Comment (compwron) [inline app/controllers/case_contacts_controller.rb:8]: It would be cleaner to put metrics methods in a different controller

**Review by compwron — APPROVED** [review]
Please add tests.

File: app/controllers/case_contacts_controller.rb (hunk 2 above)
Comment (FireLemons) [reply app/controllers/case_contacts_controller.rb:141]: No these are global metrics

**Review by littleforest — COMMENTED** [review]
@aboongm thanks for working on this! Can you update the PR title to be more descriptive. Also, could you include some screenshots in the PR?

File: app/javascript/application.js (hunk 1 above)
Comment (aboongm) [reply app/javascript/application.js]: Its updated

File: app/controllers/case_contacts_controller.rb (hunk 3 above)
Comment (aboongm) [reply app/controllers/case_contacts_controller.rb]: The method is now in health_controller.rb moved from case_contacts_controller.rb

**Comment by aboongm** [comment]
> ### What github issue is this PR for, if any?
> Resolves #4753
> 
> ### What changed, and why?
> * package.json
>   
>   * add chart
>   * add luxon
>   * chartjs-adapter-luxon
> * case_contacts_controller.rb
>   
>   * add case_contacts_creation_times_in_last_week method
> * routes.rb
>   
>   * provides route for case_contacts_creation_times_in_last_week
> * application.js
>   
>   * require display_app_metric
> * display_app_metric.js
>   
>   * This javascript file will use chart.js and create the bubble chart
> * index.html.erb (health)
>   
>   * This will render the chart on browser
> 
> ### How will this affect user permissions?
> * NA
> 
> ### How is this tested? (please write tests!) 
> * health_spec.rb (requests test for metric method)
> 
> ### Screenshots please :)
> ### Feelings gif (optional)
> What gif best describes your feeling working on this issue? https://giphy.com/ How to embed: `alt text`
> 
> ### Feedback please? (optional)
> We are very interested in your feedback! Please give us some :) https://forms.gle/1D5ACNgTs2u9gSdh9

**Review by haiderRizvii — APPROVED** [review]

**Review by littleforest — APPROVED** [review]

**Comment by littleforest** [comment]
@aboongm could you merge in main?

**Review by haiderRizvii — APPROVED** [review]

**Review by haiderRizvii — APPROVED** [review]

**Comment by compwron** [comment]
It looks like CI is failing but otherwise looks good

**Comment by compwron** [comment]
I'm not sure what's wrong with CI but I don't see anything in this PR that should break those tests- let's merge and then debug.

**Pull request merged**